### DISCUSSION TITLE
Built-in support to download TTS models en masse?

### DISCUSSION THREAD

**Discussion by AndrewBarfield** [coqui-ai/TTS#978] [General]
I'm creating a Docker container for TTS and would like to be able to download all models/vocoders in one fell swoop.

I've found: https://github.com/coqui-ai/TTS/blob/main/TTS/.models.json, so I could write a script to do it but that could get stale quickly.

Is there a command-line argument to do this for TTS or TTS-Server?

Thanks in advance!

_(Completely unrelated P.S.: Is the default voice Korva Coleman?)_

**Comment by erogol** [comment]
Default is LJSpeech dataset. 

There is no command to download all the models. You need to call once each model or you can use ModelManager class to do that manually.

**Comment by meonkeys** [comment]
@AndrewBarfield would you mind sharing your Dockerfile/project? I'm trying to modify and build https://github.com/synesthesiam/coqui-docker/ for amd64 (CPU) only but I haven't got it working yet.

**Reply by AndrewBarfield** [reply]
Hope this helps!

[Dockerfile.txt](https://github.com/coqui-ai/TTS/files/8400717/Dockerfile.txt)

**Reply by meonkeys** [reply]
thank you! Sure does. Is `fix-permissions` anything special? Maye just a recursive `chown`? I guess it is a separate script.

**Reply by AndrewBarfield** [reply]
It ensures everything is accessible via the coqui user.  Here it is:
[fix-permissions.txt](https://github.com/coqui-ai/TTS/files/8401420/fix-permissions.txt)

This general dockerfile setup/architecture (including 'fix-permissions') was taken from https://github.com/jupyter/docker-stacks.

**Comment by AndrewBarfield** [comment]
Oh, BTW, here's the script that I use to download all the models.

```bash
#!/bin/bash

list=$(tts --list_models | grep -oP 'tts_models[^\s]*')

while IFS= read -r line; do
	filename=$(basename -- "$line")
	tts --text "Hello world!" \
		--model_name "$line" \
		--out_path "wav/$filename.wav"
	echo -e "\n"
done <<< "$list"
```

**Reply by meonkeys** [reply]
Thanks again!

**Reply by meonkeys** [reply]
Have you seen https://www.web3.lu/text-to-speech-sound-samples-from-coqui-tts/ ? I found that in https://github.com/coqui-ai/TTS/discussions/522

**Reply by brandonscript** [reply]
Updated script, this one filters by language (change it from /en to whatever you want, or remove that bit from lines 3 and 11) and handles `--list_models` returning values that start with ` ##: `.

```
#!/bin/bash

list=$(tts --list_models | grep "tts_models/en/")

out_path="./path/to/output"

mkdir -p "$out_path"

while IFS= read -r line; do
        model=$(echo "$line" | sed -E 's/^ [0-9]+: //' | sed -E 's/ \[already downloaded\]$//')
	filename=$(echo "$model" | sed -E 's/^tts_models\/en\///' | sed -E 's/\//_/g')
	cmd="tts --text \"Hello world!\" \
--model_name \"$model\" \
--out_path \"$out_path/$filename.wav\""
	eval "$cmd"
done <<< "$list"
```

**Reply by ErcinDedeoglu** [reply]
> You must agree to the terms of service to use this model.
 | > Please see the terms of service at https://coqui.ai/cpml.txt
 | > "I have read, understood and agreed the Terms and Conditions." - [y/n]
 | | > Traceback (most recent call last):
  File "/usr/local/bin/tts", line 8, in <module>
    sys.exit(main())
  File "/root/TTS/bin/synthesize.py", line 424, in main
    model_path, config_path, model_item = manager.download_model(args.model_name)
  File "/root/TTS/utils/manage.py", line 403, in download_model
    self.create_dir_and_download_model(model_name, model_item, output_path)
  File "/root/TTS/utils/manage.py", line 331, in create_dir_and_download_model
    raise Exception(" [!] You must agree to the terms of service to use this model.")
Exception:  [!] You must agree to the terms of service to use this model.
